src/shaka_scheme/system/gc/GCList.cpp

```diff
@@ -4,67 +4,78 @@
 
 #include "shaka_scheme/system/gc/GCData.hpp"
 #include "shaka_scheme/system/gc/GCList.hpp"
+#include <utility>
 
 namespace shaka {
+    namespace gc {
 
-    GCList::GCList() {
-        this->size = 0;
-        this->head = nullptr;
-    }
+        GCList::GCList() {
+            this->size = 0;
+            this->head = nullptr;
+        }
 
-    GCList::~GCList() {
-        GCData *conductor = this->head;
+        GCList::~GCList() {
+            GCData *conductor = this->head;
 
-        while(conductor != nullptr) {
-            GCData *temp = conductor;
-            conductor = conductor->get_next();
-            delete temp;
-            this->size--;
+            while (conductor != nullptr) {
+                GCData *temp = conductor;
+                conductor = conductor->get_next();
+                delete temp;
+                this->size--;
+            }
+        }
+    
+        GCList::GCList(GCList&& other) : size(0), head(nullptr) {
+            swap(*this, other);
         }
-    }
 
-    bool GCList::is_empty() {
-        return this->size == 0 && this->head == nullptr;
-    }
+        bool GCList::is_empty() {
+            return this->size == 0 && this->head == nullptr;
+        }
 
-    int GCList::get_size() {
-        return this->size;
-    }
+        int GCList::get_size() {
+            return this->size;
+        }
 
-    void GCList::add_data(GCData *data) {
-        data->set_next(head);
-        this->head = data;
-        this->size++;
-    }
+        void GCList::add_data(shaka::gc::GCData *data) {
+            data->set_next(head);
+            this->head = data;
+            this->size++;
+        }
 
-    void GCList::sweep() {
-        GCData *prev = nullptr;
-        GCData *curr = this->head;
+        void GCList::sweep() {
+            GCData *prev = nullptr;
+            GCData *curr = this->head;
 
-        if(this->is_empty()) return;
+            if (this->is_empty()) return;
```

Comment:
Please add curly braces here. Having one-off lines with no braces is not good style.

Reply:
ok done!